src/bvh/optimization.rs

```diff
@@ -0,0 +1,359 @@
+use bvh::*;
```

Comment:
Docstring added in c29b791.
As for the complexity - i elsewhere mentioned that i have been generously commenting the code. Splitting the functions into smaller functions lets you run into more borrow checker issues that have to be fixed by adding even more rust-specific code. I think the gain would be pretty minimal.